Core/Core/View/Base/FullScreenErrorView/FullScreenErrorView.swift

```diff
@@ -34,56 +39,87 @@ public struct FullScreenErrorView: View {
+        switch errorType {
+        case .noVideos(error: let error, image: let image),
+                .noHandouts(error: let error, image: let image),
+                .noAnnouncements(error: let error, image: let image),
+                .noCourseDates(error: let error, image: let image),
+                .noCourseware(error: let error, image: let image):
+            GeometryReader { proxy in
+                VStack(spacing: 20) {
+                    Spacer()
+                    image
+                        .resizable()
                         .renderingMode(.template)
-                        .foregroundStyle(Color.primary)
-                        .scaledToFit()
+                        .foregroundStyle(Theme.Colors.textSecondary)
+                        .aspectRatio(contentMode: .fit)
+                        .frame(maxWidth: 72, maxHeight: 80)
                     
-                    Text(CoreLocalization.Error.Internet.noInternetTitle)
-                        .font(Theme.Fonts.titleLarge)
-                        .foregroundColor(Theme.Colors.textPrimary)
-                    
-                    Text(CoreLocalization.Error.Internet.noInternetDescription)
-                        .font(Theme.Fonts.bodyLarge)
-                        .foregroundColor(Theme.Colors.textPrimary)
-                        .multilineTextAlignment(.center)
-                        .padding(.horizontal, 50)
-                case .generic:
-                    CoreAssets.notAvaliable.swiftUIImage
-                        .renderingMode(.template)
-                        .foregroundStyle(Color.primary)
-                        .scaledToFit()
-                    
-                    Text(CoreLocalization.View.Snackbar.tryAgainBtn)
-                        .font(Theme.Fonts.titleLarge)
-                        .foregroundColor(Theme.Colors.textPrimary)
-                    
-                    Text(CoreLocalization.Error.unknownError)
-                        .font(Theme.Fonts.bodyLarge)
+                    Text(error)
+                        .font(Theme.Fonts.labelLarge)
                         .foregroundColor(Theme.Colors.textPrimary)
                         .multilineTextAlignment(.center)
                         .padding(.horizontal, 50)
+                    Spacer()
                 }
-                
-                if errorType != .noInternet {
-                    UnitButtonView(
-                        type: .reload,
-                        action: {
-                            self.action()
-                        }
-                    )
+                .frame(maxWidth: .infinity, maxHeight: proxy.size.height)
+                .background(
+                    Theme.Colors.background
+                )
+            }
+        default:
+            GeometryReader { proxy in
+                VStack(spacing: 20) {
+                    Spacer()
+                    switch errorType {
```

Comment:
It seems bit odd to use the switch again under a case of the same switch. How about using a single `GeometryReader` and configure it according to the need.

Reply:
Thank you for pointing that out. I have fixed it.